server/sonar-server/src/main/java/org/sonar/server/ce/ws/ActivityStatusAction.java

```diff
@@ -0,0 +1,127 @@
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.web.UserRole;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.core.util.Uuids;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.ce.CeActivityDto;
+import org.sonar.db.ce.CeQueueDto;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.server.component.ComponentFinder;
+import org.sonar.server.user.UserSession;
+import org.sonar.server.ws.KeyExamples;
+import org.sonarqube.ws.WsCe.ActivityStatusWsResponse;
+import org.sonarqube.ws.client.ce.ActivityStatusWsRequest;
+
+import static org.sonar.server.component.ComponentFinder.ParamNames.COMPONENT_ID_AND_KEY;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+import static org.sonarqube.ws.client.ce.CeWsParameters.PARAM_COMPONENT_ID;
+import static org.sonarqube.ws.client.ce.CeWsParameters.PARAM_COMPONENT_KEY;
+
+public class ActivityStatusAction implements CeWsAction {
+  private final UserSession userSession;
+  private final DbClient dbClient;
+  private final ComponentFinder componentFinder;
+
+  public ActivityStatusAction(UserSession userSession, DbClient dbClient, ComponentFinder componentFinder) {
+    this.userSession = userSession;
+    this.dbClient = dbClient;
+    this.componentFinder = componentFinder;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller
+      .createAction("activity_status")
+      .setDescription("Return CE activity related metrics.<br>" +
+        "Requires 'Administer System' permission or 'Administer' rights on the specified project.")
+      .setSince("5.5")
+      .setResponseExample(getClass().getResource("activity_status-example.json"))
+      .setInternal(true)
+      .setHandler(this);
+
+    action.createParam(PARAM_COMPONENT_ID)
+      .setDescription("Id of the component (project) to filter on")
+      .setExampleValue(Uuids.UUID_EXAMPLE_03);
+    action.createParam(PARAM_COMPONENT_KEY)
+      .setDescription("Key of the component (project) to filter on")
+      .setExampleValue(KeyExamples.KEY_PROJECT_EXAMPLE_001);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    ActivityStatusWsResponse activityStatusResponse = doHandle(toWsRequest(request));
+    writeProtobuf(activityStatusResponse, request, response);
+  }
+
+  private ActivityStatusWsResponse doHandle(ActivityStatusWsRequest request) {
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      Optional<ComponentDto> component = searchComponent(dbSession, request);
+      String componentUuid = component.isPresent() ? component.get().uuid() : null;
+      checkPermissions(componentUuid);
+      int pendingCount = dbClient.ceQueueDao().countByStatusAndComponentUuid(dbSession, CeQueueDto.Status.PENDING, componentUuid);
```

Comment:
I wanted to differentiate status from queue and from activity because it can be confusing